metadata-utils/src/main/java/com/linkedin/metadata/utils/metrics/MetricUtils.java

```diff
@@ -1,51 +1,83 @@
 package com.linkedin.metadata.utils.metrics;
 
-import com.codahale.metrics.Counter;
-import com.codahale.metrics.MetricRegistry;
-import com.codahale.metrics.SharedMetricRegistries;
-import com.codahale.metrics.Timer;
-import com.codahale.metrics.jmx.JmxReporter;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
 
+import java.util.HashMap;
+import java.util.Map;
 
 public class MetricUtils {
   private MetricUtils() {
   }
 
   public static final String DELIMITER = "_";
 
-  public static final String NAME = "default";
-  private static final MetricRegistry REGISTRY = SharedMetricRegistries.getOrCreate(NAME);
+  private static final PrometheusMeterRegistry PROMETHEUS_METER_REGISTRY = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+  private static final Map<String, LongTaskTimer> TIMER_MAP = new HashMap<>();
+  private static final Map<String, DistributionSummary> HISTOGRAM_MAP = new HashMap<>();
 
-  static {
-    final JmxReporter reporter = JmxReporter.forRegistry(REGISTRY).build();
-    reporter.start();
+  public static Counter counter(Class<?> klass, String... subNames) {
+    return counter(klass.toString(), subNames);
   }
 
-  public static MetricRegistry get() {
-    return REGISTRY;
+  public static Counter counter(String metricName, String... subNames) {
+    String name = buildName(metricName, subNames);
+    return PROMETHEUS_METER_REGISTRY.counter(name);
   }
 
-  public static Counter counter(Class<?> klass, String metricName) {
-    return REGISTRY.counter(MetricRegistry.name(klass, metricName));
+  public static LongTaskTimer timer(Class<?> klass, String... subNames) {
+    return timer(klass.toString(), subNames);
+  }
+
+  public static LongTaskTimer timer(String metricName, String... subNames) {
+    String name = buildName(metricName, subNames);
+    if (!TIMER_MAP.containsKey(name)) {
+      TIMER_MAP.put(name, LongTaskTimer.builder(name).register(PROMETHEUS_METER_REGISTRY));
+    }
+    return TIMER_MAP.get(name);
+  }
+
+  public static DistributionSummary histogram(Class<?> klass, String... subNames) {
```

Comment:
when I run `GMS` with this change locally, I couldn't see percentiles of the distribution other then `max`.

Reply:
currently we have: min, 50th, 75th, 90th, 95th, 98, 99th, 999th max